### ISSUE TITLE
Add Lucet runtime

### ISSUE THREAD

**Issue by PiotrSikora** [proxy-wasm/proxy-wasm-cpp-host#96]
https://github.com/bytecodealliance/lucet

**Comment by mathetake** [comment]
According to the blog here(https://www.fastly.com/blog/announcing-lucet-fastly-native-webassembly-compiler-runtime), Lucet seems like designed for 1 Wasm Vm per request model.

> Lucet can instantiate WebAssembly modules in under 50 microseconds, with just a few kilobytes of memory overhead. By comparison, Chromium’s V8 engine takes about 5 milliseconds, and tens of megabytes of memory overhead, to instantiate JavaScript or WebAssembly programs.

I'm kind of tempted to do the experimental host implementation of this model rather than just using Lucet in the current Wasm runtime interface :smile: But I see it's going to require tons of work

**Comment by PiotrSikora** [comment]
Yeah, Fastly is using Lucent that way, and Cloudflare is using V8 the same way.

But it in no way means that Lucet cannot be used with the "on event" callback model that we're using in Proxy-Wasm.
